src/main/webapp/app/delete-dialog/delete-dialog-service.ts

```diff
@@ -0,0 +1,28 @@
+import { Injectable } from '@angular/core';
+import { NgbModal } from '@ng-bootstrap/ng-bootstrap';
+import { DeleteDialogComponent } from 'app/delete-dialog/delete-dialog.component';
+
+/**
+ * Data that will be passed to the delete dialog component
+ */
+export class DeleteDialogData {
+    entityTitle: string;
+    deleteQuestion: string;
+    deleteConfirmationText?: string;
+}
+@Injectable({ providedIn: 'root' })
+export class DeleteDialogService {
+    constructor(private modalService: NgbModal) {}
+
+    /**
+     * Opens delete dialog and returns a result after dialog is closed
+     * @param deleteDialogData data that is used in dialog
+     */
+    openDeleteDialog(deleteDialogData: DeleteDialogData): Promise<any> {
```

Comment:
Just as a side not as this could be unintended: It is possible to open multiple delete dialogs at the same time.

Reply:
Yes, now I hold a reference on the service